bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/ApiBridgeHandler.java

```diff
@@ -352,22 +354,20 @@ public synchronized <T> T executeUri(URI uri, HttpMethod method, Class<T> clazz,
             throw exception;
         } catch (NetatmoException e) {
             if (e.getStatusCode() == ServiceError.MAXIMUM_USAGE_REACHED) {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
-                prepareReconnection(null, null);
+                prepareReconnection(API_LIMIT_INTERVAL_S,
+                        "@text/maximum-usage-reached [ \"%d\" ]".formatted(API_LIMIT_INTERVAL_S), null, null);
             }
             throw e;
-        } catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
-            throw new NetatmoException("Request interrupted");
-        } catch (TimeoutException | ExecutionException e) {
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            if (e instanceof InterruptedException) {
+                Thread.currentThread().interrupt();
```

Comment:
I have a suspicion that the handling of `InterruptedException` could be the root cause of two of the findings in #16485, but I did not have time yesterday to confirm. I will have a look within the next few days. I also need to have a closer look at all the logs generated yesterday, as I might have missed some details in the created issue.

Do you think a recursive call of this method should take place upon `InterruptedException`? I'm wondering how the flow will look like when disposing a handler while an HTTP call is currently blocking.

Reply:
`InterruptedException` is thrown when the thread is interrupted, or perhaps more accurately when requested to be interrupted. I think there is a good chance that this could be the trigger (line 121):

https://github.com/openhab/openhab-addons/blob/ba06c0d13890a127c258d87ef2587cb956a54fed/bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/capability/RefreshCapability.java#L120-L124

I have added trace logging here in my production system now to observe how this may be connected.

I think I need to read it carefully a few times to understand it, but as an example, with a delay of 1, this will asynchronously run this method in a thread:

https://github.com/openhab/openhab-addons/blob/ba06c0d13890a127c258d87ef2587cb956a54fed/bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/capability/RefreshCapability.java#L82-L98

which will then cancel and reschedule itself in the last line. I'm wondering if any HTTP calls could be blocking at any time when this rescheduling it happening? At least that would explain the `InterruptedException`. In such case I guess it should not keep doing things when it has been requested to stop.

Just brainstorming here for now, I don't have any evidence or overview of the logic.

Reply:
> In such case I guess it should not keep doing things when it has been requested to stop.

This reminds me of an old discussion I had with @lolodomo. IIRW it breaks something else when using `job.cancel(false)`. I'm going to do some tests.

Reply:
`job.cancel(true)` is perfectly fine to avoid any lingering threads. However, the thread must also respect when being interrupted and handle `InterruptedException` accordingly. I'm also thinking to run some tests putting a `Thread.sleep(60000)` inside `executeUri` to see what happens during normal flow and when pausing the Thing.

Reply:
Got it  [here](https://github.com/openhab/openhab-addons/pull/12357#issuecomment-1113420982)

Reply:
> `job.cancel(true)` is perfectly fine to avoid any lingering threads. However, the thread must also respect when being interrupted and handle `InterruptedException` accordingly. I'm also thinking to run some tests putting a `Thread.sleep(60000)` inside `executeUri` to see what happens during normal flow and when pausing the Thing.

And that's probably the reason why InterruptedException was not suject to retry requesting ! I'll revert this.